Don't attempt to read password on stdin if stdin isn't a tty

mrmr1993 opened the pull request:
Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

Closes #6681

nholland94 commented:
Even if stdin is a tty, I think we still might want to just skip this prompt. I believe @lk86 was saying that this prompt gets hidden from users by other logs and that users end up not seeing the prompt. I'll defer to @lk86 to confirm what we should/shouldn't do.

lk86 commented:
We're always instructing people to either use an environment variable or -block-producer-password, so if we don't have a password it should always fail and point people to CODA_PRIVKEY_PASS + -block-producer-password.

mrmr1993 commented:
Updated. I've added a new function that only checks the environment variable, and changed the calls from the `daemon` command's sartup to go through that.

The `client`/`accounts`/etc. commands that require a password will still prompt if needed, and I don't this this causes us the same issues, so leaving the old functionality available to be used there.
